app/integration_action.go

```diff
@@ -244,6 +244,11 @@ func (a *App) DoPostActionWithCookie(postId, actionId, userId, selectedOption st
 			RootId:    rootPostId,
 			UserId:    userId,
 		}
+
+		if response.SkipSlackParsing {
```

Comment:
I would love to have this `if` little bit higher, so we will not call `model.ParseSlackLinksToMarkdown` when not needed.

Reply:
+1 on the second version you propose @someone-somenet-org

Reply:
Duplication is a bad idea, second one is way better, but what you can do is first check the `response.SkipSlackParsing` to initialize just the message and after that initialize the response with the correct message.

Reply:
adapted. see new commit.

Reply:
@someone-somenet-org this is clearly better. I don't want to be a pain, but I don't like `Post` object been half-baked on creation. What do you think about something like this?
```go
message := response.EphemeralText
if !response.SkipSlackParsing {
	message = model.ParseSlackLinksToMarkdown(message)
}
ephemeralPost := &model.Post{
	Message:   message,
	ChannelId: upstreamRequest.ChannelId,
	RootId:    rootPostId,
	UserId:    userId,
}
```

Reply:
This looks ugly imho. I dont like to create new vars if it can be obviously avoided. Also it imho adds visual complexity where there isnt.

What about... ?
 ```go
 ephemeralPost := &model.Post{
 	Message:   response.EphemeralText,
 	ChannelId: upstreamRequest.ChannelId,
 	RootId:    rootPostId,
 	UserId:    userId,
 }

 if !response.SkipSlackParsing {
 	ephemeralPost.message = model.ParseSlackLinksToMarkdown(response.EphemeralText)
 }
 ```

Reply:
Way better, thanks @someone-somenet-org 👍

Reply:
done, I guess.